pkg/provisioner/crdprovisioner.go

```diff
@@ -170,45 +170,39 @@ func (c *CrdProvisioner) CreateVolume(
 	azVolumeInstance, err := azV.Get(ctx, strings.ToLower(validVolumeName), metav1.GetOptions{})
 
 	if err == nil {
-		if azVolumeInstance.Status != nil {
-			if azVolumeInstance.Status.Error != nil {
-				updatedInstance := azVolumeInstance.DeepCopy()
-				// If the CreateVolume call is being retried after erroring out previously, set
-				// the status to nil to retrigger volume creation
-				updatedInstance.Status = nil
-				azVolumeUpdated, err := azV.UpdateStatus(ctx, updatedInstance, metav1.UpdateOptions{})
-				if err != nil || azVolumeUpdated == nil || azVolumeUpdated.Status != nil {
-					klog.Errorf("Failed to update azvolume resource for volume name (%s), error: %v", volumeName, err)
+		if azVolumeInstance.Status.Detail != nil {
+			// If current request has different specifications than the existing volume, return error.
+			if !isAzVolumeSpecSameAsRequestParams(azVolumeInstance, maxMountReplicaCount, capacityRange, parameters, secrets, volumeContentSource, accessibilityReq) {
+				return nil, status.Errorf(codes.AlreadyExists, "Volume with name (%s) already exists with different specifications", volumeName)
+			}
+			// The volume creation was successful previously,
+			// Returning the response object from the status
+			return azVolumeInstance.Status.Detail.ResponseObject, nil
+		} else if azVolumeInstance.Status.Error != nil {
+			updatedInstance := azVolumeInstance.DeepCopy()
+			updatedInstance.Status.Error = nil
+			updatedInstance.Status.State = v1alpha1.VolumeOperationPending
```

Comment:
Shouldn't we make a  azV.UpdateStatus call here after updating the status of the instance?

Reply:
You are right but a regular azV.Update call should do because status is no longer a subresource.

So, we could pull the update call in line 194 out of the if clause.